components/lib/dropdown/Dropdown.js

```diff
@@ -588,12 +588,24 @@ export const Dropdown = React.memo(
             return isOptionDisabled(option) ? findPrevOption(i) : option;
         };
 
+        const findInArray = (visibleOptions, searchText) => {
+            if (!searchText || !visibleOptions?.length) return -1;
+
+            const normalizedSearch = searchText.toLocaleLowerCase();
+
+            const exactMatch = visibleOptions.findIndex((item) => getOptionLabel(item).toLocaleLowerCase() === normalizedSearch);
+
+            if (exactMatch !== -1) return exactMatch;
+
+            return visibleOptions.findIndex((item) => getOptionLabel(item).toLocaleLowerCase().startsWith(normalizedSearch));
+        };
```

Comment:
Won't this do it more efficientialy by only calling FindIndex 1 instead of passing over the whole array twice?

```js
const search = (visibleOptions, searchText) => {
    if (!searchText || !visibleOptions?.length) return -1;

    const normalizedSearch = searchText.toLocaleLowerCase();

    return visibleOptions.findIndex((item) => {
        const label = getOptionLabel(item).toLocaleLowerCase();
        return label === normalizedSearch || label.startsWith(normalizedSearch);
    });
};
```

I would also call the method `search` instead of `findInArray`

Reply:
Oh no i see you want to search the whole list for an exact match before trying again with `startswith`

Reply:
Right, we will need to iterate the array twice in this case. Once to find the exact match and then to find matches with startsWith.